src/generate/utils/State_AI.cpp

```diff
@@ -7,44 +7,58 @@
 #include "State_AI.hpp"
 #include "generate/OperationGenerator.hpp"
 #include "execute/OperationExecutor.hpp"
+#include <spdlog/spdlog.h>
 
 namespace spy::gameplay {
 
-    std::vector<State_AI> State_AI::getSuccessorStates(const util::UUID &characterId,
-                                                       const spy::MatchConfig &config,
-                                                       const libclient::LibClient &libClient) {
-        std::vector<State_AI> successors;
+    void State_AI::getSuccessorStates(const util::UUID &characterId,
+                                      const spy::MatchConfig &config,
+                                      const libclient::LibClient &libClient,
+                                      std::vector<State_AI> &successors) {
         this->wasHoneyTrapUsed = false;
         auto operations = OperationGenerator::generate(*this, characterId, config);
 
         for (const auto &op: operations) {
             auto successorStates = OperationExecutor::execute(*this, op, config, libClient);
             successors.insert(successors.end(), successorStates.begin(), successorStates.end());
         }
-
-        return successors;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f20-for-out-output-values-prefer-return-values-to-output-parameters

Reply:
Ich weiß, aber so spare ich mir Kopier- und Einfügevorgänge und somit Zeit